### PR TITLE
Fixes:

### PR DESCRIPTION
**Pull request by carioca-au**
- improved log
- created function to give extra time to io save process

### CODE REVIEW

File: deafrica/data/gmw.py [hunk 1]
```diff
@@ -73,10 +74,8 @@ def create_and_upload_stac(cog_file: Path, s3_dst: str, year) -> Item:
     )
 
     log.info("assets creation")
-    del item.assets["asset"]
-
     out_data = out_path / cog_file.name
-    item.assets["mangrove "] = pystac.Asset(
+    item.assets["asset"] = pystac.Asset(
```
Comment (alexgleith) [inline deafrica/data/gmw.py]: I think you should be doing this using the arguments for the `create_stac_item` function

File: deafrica/data/gmw.py [hunk 2]
```diff
@@ -138,28 +138,26 @@ def gmw_download_stac_cog(year: str, s3_dst: str) -> None:
         )
         check_output(cmd, stderr=STDOUT, shell=True)
 
+        io_timer(file_path=output_file, log=log)
```
Comment (alexgleith) [inline deafrica/data/gmw.py:143]: WHy not just log the time?

File: deafrica/utils.py [hunk 3]
```diff
@@ -208,6 +210,19 @@ def test_http_return(returned):
...
+    """
+    Function to check if file exist.
+    """
+    # give extra time to output write the file in disk
+    count = 0
+    while not os.path.exists(file_path) and count < 4:
+        if log:
+            log.info(f"Wait OS write file in disk {count}s")
+        time.sleep(1)
+        count += 1
```
Comment (alexgleith) [inline deafrica/utils.py:223]: I don't get this

**Review by alexgleith — DISMISSED** [review]
What are you doing with the io thing?

File: deafrica/data/gmw.py (hunk 1 above)
Comment (carioca-au) [reply deafrica/data/gmw.py]: I am doing that:
https://github.com/digitalearthafrica/deafrica-scripts/blob/518d9457508917aaf42a84d50916b2be2ad6ec33/deafrica/data/gmw.py#L50

Comment (carioca-au) [reply deafrica/data/gmw.py]: I'm also following your steps:
https://github.com/digitalearthafrica/deafrica-scripts/blob/518d9457508917aaf42a84d50916b2be2ad6ec33/deafrica/data/chirps.py#L72

Comment (alexgleith) [reply deafrica/data/gmw.py]: I think I did it the wrong way.

Pass in `asset_name="mangrove"` and `asset_roles=["data"]` and `asset_href=...`

File: deafrica/utils.py (hunk 3 above)
Comment (carioca-au) [reply deafrica/utils.py:223]: @NikitaGandhi and I found out that the main issue in Argo was that when the second task tried to get the file, the file wasn't in the folder yet. It means that Argo was faster to execute the lines than write the file on disk.
The way to solve that was by adding a sleep(1.5), which @NikitaGandhi asked me to add a file check instead a sleep, so I added this function which checks the file and if not there sleeps a bit and checks again. I added a limit to never be in an infinit loop.

File: deafrica/data/gmw.py (hunk 2 above)
Comment (carioca-au) [reply deafrica/data/gmw.py:143]: This function isn't logging the time, it's giving Argo time to write the file before go ahead.

File: deafrica/data/gmw.py (hunk 1 above)
Comment (carioca-au) [reply deafrica/data/gmw.py]: This was the way that I was doing before and you asked me to change lol..... I'll roll back.

File: deafrica/data/gmw.py (hunk 2 above)
Comment (alexgleith) [reply deafrica/data/gmw.py:143]: Argo? This doesn't make sense at all.

File: deafrica/utils.py (hunk 3 above)
Comment (alexgleith) [reply deafrica/utils.py:223]: It shouldn't be necessary, but ok.

Comment (carioca-au) [reply deafrica/utils.py:223]: I totally agree with you, and I was totally against this solution when Nikita asked me to try, however, we tried and worked ..... We couldn't find any other explanation.
As always I'm open to other and better options.

File: deafrica/data/gmw.py (hunk 2 above)
Comment (carioca-au) [reply deafrica/data/gmw.py:143]: I said Argo but I know that Argo just executes the container, what I meant was, somehow when the process executes the second task, **rio cogeo create**, it seems that the file isn't available yet.

File: deafrica/utils.py (hunk 3 above)
Comment (carioca-au) [reply deafrica/utils.py:223]: PS=> Locally running in my Docker, the process worked just fine without the use of this function.

**Review by NikitaGandhi — APPROVED** [review]

**Pull request merged**